pkg/discovery/network/flannel_test.go

```diff
@@ -63,10 +63,13 @@ var flannelDaemonSet = appsv1.DaemonSet{
 	ObjectMeta: metav1.ObjectMeta{
 		Name:      "kube-flannel-ds",
 		Namespace: metav1.NamespaceSystem,
+		Labels:    map[string]string{"k8s-app": "flannel"},
```

Comment:
Should add a test case for DaemonSet without flannel label returning a generic plugin.

Reply:
Added the test 

<img width="1001" alt="image" src="https://github.com/user-attachments/assets/bf375cec-995b-4bc6-91c7-666a32c8aee2">